### PR TITLE
[JSON RPC] Refactor set_response_error()

### PR DESCRIPTION
**Pull request by JoshLind**
## Motivation

This small PR refactors some of the error handling code in the JSON RPC server. More specifically, it refactors counter labels into constants and provides a helper method to set the JSON RPC response error (to minimize code duplication).

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

All local tests still pass.

## Related PRs

None.

### CODE REVIEW

File: json-rpc/src/runtime.rs [hunk 1]
```diff
@@ -19,6 +19,14 @@ use warp::{
     Filter,
 };
 
+// Counter labels for runtime metrics
+const LABEL_FAIL: &str = "fail";
```
Comment (mimoo) [inline json-rpc/src/runtime.rs:23]: maybe a more descriptive error :P ?

**Comment by mimoo** [comment]
I'm surprised that this actually adds more lines than it removes, but it makes it easier to read the function so it's a win in my book

File: json-rpc/src/runtime.rs (hunk 1 above)
Comment (JoshLind) [reply json-rpc/src/runtime.rs:23]: I'm gonna punt on this for now. I'm not sure how much ecosystem support there already is for the JSON RPC metrics, so will refrain from modifying the counter name. I agree though.. it's not great!

**Comment by JoshLind** [comment]
Thanks, David! 😄 @bors-libra r=mimoo

> I'm surprised that this actually adds more lines than it removes, but it makes it easier to read the function so it's a win in my book

Yeah.. I think it's because I added constants 😄

**Pull request closed without merging**